Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -16,23 +24,48 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+
+    return ampHour_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
+
+    if(current_<0){
+
+        return true;
+    }
     return false;
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
-{
-    return QTime::currentTime();
+{    QTime charging;
+     charging.setHMS(0,0,0,0);
+
+    if(isCharging()==true){
```

Comment:
you don't need to include the check to true, if you simply call a boolean function within a test expression the returned value will be checked accordingly :)